### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by victor-carvalho**
In this PR I update most of the outdated dependencies of pitaya and also make the integration tests more easy to run on OSX (no more popup allow spam). I tried to update every dependency but it would need a lot of changes in the code because of etcd dependency that not only changed the package but also the structure of the packages.

### CODE REVIEW

File: cluster/grpc_rpc_server.go [hunk 1]
```diff
@@ -52,8 +52,9 @@ func NewGRPCServer(config *config.Config, server *Server, metricsReporters []met
 
 // Init inits grpc rpc server
 func (gs *GRPCServer) Init() error {
+	address := gs.config.GetString("pitaya.cluster.rpc.server.grpc.address")
```
Comment (leohahn) [inline cluster/grpc_rpc_server.go:55]: won't address always be 0.0.0.0?

**Comment by leohahn** [comment]
Does this  PR solve this issue? https://github.com/topfreegames/pitaya/issues/147

File: cluster/grpc_rpc_server.go (hunk 1 above)
Comment (victor-carvalho) [reply cluster/grpc_rpc_server.go:55]: No, in the tests when starting the I pass the `-address` flag with `localhost` value.

**Comment by victor-carvalho** [comment]
> Does this PR solve this issue? #147

No, to update etcd would need some changes in the code so I decided to do this in another separate PR. I don't know really know the extent of the changes as in newer versions the etcd package structure changed.

**Comment by leohahn** [comment]
Fair enough, apart from the comment LGTM

**Review by rodopoulos — APPROVED** [review]
LGTM

File: cluster/grpc_rpc_server.go (hunk 1 above)
Comment (leohahn) [reply cluster/grpc_rpc_server.go:55]: oh ,oks

**Review by leohahn — APPROVED** [review]

**Review by gabrielcorado — APPROVED** [review]

**Pull request merged**